File: Seunghwan/week7/BOJ11047.py
n, k = list(map(int, input().split())) 
coin_types = [] # 종류들
count = 0

for i in range(n):
    coin_types.append(int(input()))

for coin in reversed(range(n)):
    count += k // coin_types[coin] # 카운트 값 + k를 동전으로 나눈 몫
    k %= coin_types[coin]
    # 동전으로 나눈 나머지로 반복

print(count)

File: Hanna/Week7/BOJ11047.js
const { count } = require('console');
const fs = require('fs');
const { format } = require('path');
const stdin = (process.platform === 'linux'
  ? fs.readFileSync('/dev/stdin').toString()
  : `10 5000
1
5
10
50
100
500
1000
5000
10000
50000
`
).split('\n');

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

let [n,k] = input().split(' ').map(Number);
let kind = n;
let won = k;
let kindOf = [];
let i=0;
let min=0;

while(n--)
{
    kindOf[i++] = parseInt(input());
}

for(i = kind-1; i >= 0; i--)
{
    if(kindOf[i] > won){
        continue;
    }else{
        min += Math.floor(won / kindOf[i]);
        won -= (Math.floor(won / kindOf[i]) * kindOf[i]);
    }
}

console.log(min);